json_visitor.go

```diff
@@ -0,0 +1,278 @@
+package raymond
+
+import (
+	"encoding/json"
+	"github.com/mailgun/raymond/v2/ast"
+	"strings"
+)
+
+type list []interface{}
+
+func (l *list) Add(item interface{}) {
+	*l = append(*l, item)
+}
+
+func newList(item interface{}) *list {
+	l := new(list)
+	l.Add(item)
+	return l
+}
+
+type JSONVisitor struct {
+	JSON map[string]interface{}
+	ctx  *handlebarsContext
+}
+
+func newJSONVisitor() *JSONVisitor {
+	j := map[string]interface{}{}
+	v := &JSONVisitor{JSON: j, ctx: newHandlebarsContext()}
+	return v
+}
+
+func ToJSON(node ast.Node) string {
+	visitor := newJSONVisitor()
+	node.Accept(visitor)
+	b, _ := json.Marshal(visitor.JSON)
+	return string(b)
+}
+
+func (v *JSONVisitor) VisitProgram(node *ast.Program) interface{} {
+	for _, n := range node.Body {
+		n.Accept(v)
+	}
+	return v.JSON
+}
+
+func (v *JSONVisitor) VisitMustache(node *ast.MustacheStatement) interface{} {
+
+	node.Expression.Accept(v)
+
+	return nil
+}
+
+func (v *JSONVisitor) VisitBlock(node *ast.BlockStatement) interface{} {
+	var action string
+	fp := node.Expression.FieldPath()
+	if fp != nil {
+		action = fp.Parts[0]
```

Comment:
Is it possible for `fp.Parts` to have zero elements? This could cause a panic. Maybe we use `node.Expression.HelperName()` instead to fetch the `action`? i think `HelperName == action`.

Reply:
yeeee good call.